index.js

```diff
@@ -1,23 +1,30 @@
 /*!
  * collection-map <https://github.com/jonschlinkert/collection-map>
  *
- * Copyright (c) 2015, Jon Schlinkert.
- * Licensed under the MIT License.
+ * Copyright (c) 2015, 2017, Jon Schlinkert.
+ * Released under the MIT License.
  */
 
 'use strict';
 
-var isObject = require('is-plain-object');
-var iterator = require('make-iterator');
-var values = require('get-values');
+var makeIterator = require('make-iterator');
+var forOwn = require('for-own');
 var map = require('arr-map');
 
-module.exports = function(collection, cb, thisArg) {
-  cb = iterator(cb, thisArg);
-  if (isObject(collection) && !collection.length) {
-    collection = values(collection);
+module.exports = function(collection, fn, thisArg) {
+  if (!Array.isArray(collection)) {
+    var iterator = makeIterator(fn);
+    var result = [];
+
+    forOwn(collection, function(value, key) {
+      if (typeof fn === 'string') {
+        console.log(collection[fn])
```

Comment:
oh I thought you'd like for me to leave some Easter eggs in the code. jk lol. thx I got distracted when my girls got home earlier.